src/Raven.Server/Documents/Indexes/Workers/MapItems.cs

```diff
@@ -103,12 +103,21 @@ protected MapItems(Index index, IndexStorage indexStorage, MapReduceIndexingCont
                                     try
                                     {
                                         current.SkipLuceneDelete = _indexStorage.LowerThanLastDatabaseEtagOnIndexCreation(current.Etag);
-                                        var numberOfResults = _index.HandleMap(current, mapResults,
-                                            writeOperation, indexContext, collectionStats);
-
-                                        resultsCount += numberOfResults;
-                                        collectionStats.RecordMapSuccess();
-                                        _index.MapsPerSec?.MarkSingleThreaded(numberOfResults);
+                                        if (itemEnumerator.Current.DocumentFlags.HasFlag(DocumentFlags.Archived))
+                                        {
+                                            if (_index.MustDeleteArchivedDocument(current))
+                                                _index.HandleDelete(new Tombstone { LowerId = itemEnumerator.Current.LowerId }, collection, writeOperation, indexContext, stats);
```

Comment:
Let's move it to a method like `_index.HandleArchived()`. I'm pretty sure the code handling archived docs will become more complex.

Reply:
Done